### PR TITLE
feat(server): Add maxPodLogsToRender setting (#9131)

### LINKED ISSUE #9131: Make maxPodLogsToRender configurable
# Summary
I have an Issue with showing k8s deployment logs, I found that the error is the logs of deployment can not be displayed for a certain amount of pod for example I have 20 pods in one deployment when I try to show the logs of each pod it appears but when I try to get the logs of the whole deployment I can not and I get the below error:
rpc error: code = Unknown desc = Max pods to view logs are reached. Please provide more granular query.

# Proposal
Can this value `"maxPodLogsToRender"` be configurable.

Thanks.

### PR DESCRIPTION
**Pull request by lukaszgyg**
This PR is adding the configurable `maxPodLogsToRender `. This will allow the Argo-CD administrators to allow higher number of pods logs to be rendered in UI and avoid the `rpc error: code = Unknown desc = Max pods to view logs are reached. Please provide more granular query.` error as described in the [issue #9131](https://github.com/argoproj/argo-cd/issues/9131).
Closes [ISSUE #9131](https://github.com/argoproj/argo-cd/issues/9131)

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: server/application/application.go [hunk 1]
```diff
@@ -1569,8 +1569,13 @@ func (s *Server) PodLogs(q *application.ApplicationPodLogsQuery, ws application.
 		return nil
 	}
 
-	if len(pods) > maxPodLogsToRender {
-		return errors.New("Max pods to view logs are reached. Please provide more granular query.")
```
Comment (alexef) [inline server/application/application.go:1573]: this is the root error that ends up in the argocd throwing 500 error responses. 

while I agree having a configurable limit may mitigate this to some extent, can we also solve the generic case and treat this error differently?

a 4xx response may be more fitted.

Comment (lukaszgyg) [reply server/application/application.go:1573]: As the specifying the log query will fix the problem I took the `codes.InvalidArgument` which will be transformed to `http.StatusBadRequest`

**Comment by lukaszgyg** [comment]
@leoluz I [updated](https://github.com/argoproj/argo-cd/blob/40dca79ef9b283bdf74c9645837f94e362deb627/docs/operator-manual/argocd-cm.yaml#L321) my PR base on your suggestion from the ArgoCD contributor meeting. Can you take a look?

File: server/application/application_test.go [hunk 2]
```diff
@@ -1965,6 +1965,60 @@ func TestLogsGetSelectedPod(t *testing.T) {
 	})
 }
 
+func TestMaxPodLogsRender(t *testing.T) {
```
Comment (crenshaw-dev) [inline server/application/application_test.go:1968]: I think this test probably needs to cover additional cases:

1) log query under max works
2) increasing max allows a request w/ higher number of pods to work
3) new max still prevents queries which break the limit

**Review by crenshaw-dev — DISMISSED** [review]

File: server/application/application_test.go (hunk 2 above)
Comment (lukaszgyg) [reply server/application/application_test.go:1970]: @crenshaw-dev any other open points on this PR from your side?

**Comment by Naist4869** [comment]
Is there any progress? This will be my motivation to upgrade the ArgoCD version.

**Comment by lukaszgyg** [comment]
> Is there any progress? This will be my motivation to upgrade the ArgoCD version.

The implementation is done. I'm waiting for review.
@crenshaw-dev can you have a look?

File: server/application/application.go (hunk 1 above)
Comment (Jellyfrog) [reply server/application/application.go:1566]: Maybe the variable should be removed here also?
https://github.com/argoproj/argo-cd/blob/b12630c4be3dc73bc89eb5295ef3a582c7ae628b/server/application/application.go#L68

Comment (lukaszgyg) [reply server/application/application.go:1574]: Thanks @Jellyfrog for a good catch! I removed unused const.

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Pull request merged**